src/applications/edu-benefits/components/SchoolSelectField.jsx

```diff
@@ -200,92 +258,106 @@ export class SchoolSelectField extends React.Component {
               }
             </div>
             }
-            {showInstitutionsLoading && <div>
+            {showSearchResults && showInstitutionsLoading && <div>
               <LoadingIndicator message={`Searching ${institutionQuery}...`}/>
             </div>
             }
-            {showNoResultsFound && <div className="no-results-box">
+            {showSearchResults && showNoResultsFound && <div className="no-results-box">
               <p>
                 <strong>
                   {'We can’t find your school'}
                 </strong><br/>
                 {'We’re sorry. We can’t find any school that matches your entry. Please try entering a different school name or city. Or, you can check the box to enter your school information yourself.'}
               </p>
             </div>}
-            {showPaginationLoading && <div>
+            {showSearchResults && showPaginationLoading && <div>
               <LoadingIndicator message={`Loading page ${currentPageNumber} results for ${institutionQuery}...`}/>
             </div>
             }
           </div>
-          {showPagination && <Pagination
-            page={currentPageNumber} pages={pagesCount} onPageSelect={this.handlePageSelect}/>
-          }
+          {showSearchResults && showPagination && <Pagination
+            page={currentPageNumber} pages={pagesCount} onPageSelect={this.handlePageSelect}/>}
         </div>
       </fieldset>
     );
   }
 }
 
-const mapStateToProps = (state, props) => {
+const mapStateToProps = (state, ownProps) => {
   const currentPageNumber = selectCurrentPageNumber(state);
-  const facilityCodeSelected = props.formData ? props.formData : '';
+
+  const errorMessages = selectFacilityCodeErrorMessages(ownProps);
+  const facilityCodeSelected = ownProps.formData ? ownProps.formData.facilityCode : '';
   const institutionQuery = selectInstitutionQuery(state);
   const institutions = selectInstitutions(state);
   const institutionSelected = selectInstitutionSelected(state);
+  const manualSchoolEntryChecked = selectManualSchoolEntryChecked(state);
   const pagesCount = selectPagesCount(state);
   const searchInputValue = selectSearchInputValue(state);
   const searchResultsCount = selectSearchResultsCount(state);
+  const showErrors = errorMessages.length > 0 && selectFormSubmitted(ownProps) && !manualSchoolEntryChecked;
```

Comment:
Putting this kind of logic in `mapStateToProps` is kind of a bad pattern- haven't really figured out and idiomatic way to handle making it easy to extract `ownProps` + `state` stuff into easily testable functions. They're not "selectors."